src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java

```diff
@@ -64,112 +71,249 @@ public MuleConfigGenerator(List<ApikitMainFlowContainer> apis, List<GenerationMo
+    muleConfigs.add(mainMuleConfig);
+    if (apikitConfig != null) {
+      muleConfigs.add(apikitConfig);
     }
+    return muleConfigs;
+  }
+
+  /**
+   * Generates new flows based on the generation models previously created. Also updates apikit configuration in case
+   * needed.
+   * @param flowEntry new flow to be generated
+   */
+  private void generateForFlowEntries(GenerationModel flowEntry) {
+    Element apikitFlowScope = new APIKitFlowScope(flowEntry, isMuleEE()).generate();
+    MuleConfig mainMuleConfig = flowEntry.getApi().getMuleConfig();
+    mainMuleConfig.getContentAsDocument().getRootElement().getContent().add(apikitFlowScope);
+    mainMuleConfig.addFlow(new Flow(apikitFlowScope));
   }
 
-  private int getLastFlowIndex(Document doc) {
-    int lastFlowIndex = 0;
-    for (int i = 0; i < doc.getRootElement().getContentSize(); i++) {
-      Content content = doc.getRootElement().getContent(i);
-      if (content instanceof Element && "flow".equals(((Element) content).getName())) {
-        lastFlowIndex = i;
-      }
+  /**
+   * It looks for the first (and only) apikit router configuration reference within the mainMuleConfig
+   * received by parameter. Then with that it looks inside all of the files of the app for the file which
+   * calls that reference.
+   * @param mainMuleConfig the configuration which contains the reference to the apikit configuration.
+   * @return the file on which apikit configuration is made
+   */
+  private MuleConfig retrieveApikitConfigMuleConfig(MuleConfig mainMuleConfig) {
+    if (muleConfigsInApp.size() > 1) {
+      String configRef = mainMuleConfig.getMainFlows().stream().findFirst().get().getApikitRouter().getConfigRef();
+      Stream<MuleConfig> apikitConfigurations = muleConfigsInApp.stream().filter(muleConfig -> muleConfig.getContentAsDocument()
+          .getRootElement().getChild("config", APIKitTools.API_KIT_NAMESPACE.getNamespace()) != null);
+      return apikitConfigurations.filter(configuration -> filterConfigurations(configRef, configuration)).findFirst()
+          .orElse(mainMuleConfig);
     }
-    return lastFlowIndex;
+    return mainMuleConfig;
   }
 
-  private void updateApikitConfig(ApikitMainFlowContainer api, MuleConfig config) {
-    Element apikitConfiFromMuleConfig =
-        config.getContentAsDocument().getRootElement().getChild("config", APIKitTools.API_KIT_NAMESPACE.getNamespace());
-    Element apikitConfigFromApi = api.getConfig().generate();
+  /**
+   * Given an apikit configuration reference and a configuration, it looks for the file which contains
+   * the apikit configuration.
+   * @param configRef apikit configuration reference
+   * @param configuration file from the app - could contain apikit configuration
+   * @return true/false depending on existence of any file matching the configuration.
+   */
+  private boolean filterConfigurations(String configRef, MuleConfig configuration) {
```

Comment:
Update the name of the method and the javadoc to reflect that this is a validation/check, it doesn't filter or return anything.